CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc

```diff
@@ -204,12 +204,23 @@ CaloTPGTranscoderULUT::getOutputLUTSize(const HcalTrigTowerDetId& id) const
       case HcalTopologyMode::TriggerMode_2016:
          return QIE8_OUTPUT_LUT_SIZE;
       case HcalTopologyMode::TriggerMode_2017:
+         if (id.ietaAbs() <= theTopology->lastHERing())
+            return QIE8_OUTPUT_LUT_SIZE;
+         else
+            return QIE10_OUTPUT_LUT_SIZE;
+      case HcalTopologyMode::TriggerMode_2017plan1:
```

Comment:
hi @matz-e - for the next iteration - a small lookup table that lets you rewrite all this code as return lutSIze_[id.ietaAbs];  might be worth considering.